cardinal/server2/tx.go

```diff
@@ -0,0 +1,132 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/gofiber/fiber/v2"
+	"github.com/rotisserie/eris"
+	"github.com/rs/zerolog/log"
+	"pkg.world.dev/world-engine/cardinal/ecs"
+	"pkg.world.dev/world-engine/cardinal/types/message"
+	"pkg.world.dev/world-engine/sign"
+)
+
+func (handler *Handler) registerTxHandler() error {
+	world := handler.w
+	txs, err := world.ListMessages()
+	if err != nil {
+		return err
+	}
+
+	txNameToTx := make(map[string]message.Message)
+	for _, tx := range txs {
+		txNameToTx[tx.Name()] = tx
+	}
+
+	gameHandler := func(c *fiber.Ctx) error {
+		requestBody := c.Body()
+		if len(requestBody) == 0 {
+			return fiber.NewError(fiber.StatusBadRequest, "request body was empty")
+		}
+		body, sp, err := handler.getBodyAndSignedPayloadFromRequest(requestBody, false)
+		if err != nil {
+			return fiber.NewError(fiber.StatusBadRequest, err.Error())
+		}
+		tx, err := getTxFromParams(c, txNameToTx)
+		if err != nil {
+			return fiber.NewError(fiber.StatusNotFound, err.Error())
+		}
+		txReply, err := handler.processTransaction(tx, body, sp)
+		if err != nil {
+			return fiber.NewError(fiber.StatusInternalServerError, err.Error())
+		}
+		return c.JSON(txReply)
+	}
+
+	createPersonaHandler := func(c *fiber.Ctx) error {
+		requestBody := c.Body()
+		body, sp, err := handler.getBodyAndSignedPayloadFromRequest(requestBody, true)
+		if err != nil {
+			if eris.Is(err, eris.Cause(ErrInvalidSignature)) || eris.Is(err, eris.Cause(ErrSystemTransactionRequired)) ||
+				eris.Is(err, eris.Cause(ErrNonceVerificationFailed)) || eris.Is(err, eris.Wrap(errors.Join(ErrInvalidSignature, ErrSignatureValidationFailed), "")) {
```

Comment:
Looking for suggestions on how to not use `eris.Wrap(errors.Join(ErrInvalidSignature, ErrSignatureValidationFailed)`. The tests fail without specifically adding that line since it comes from the `verifySignature` function. Using `eris.Is(err, ErrInvalidSignature)` by itself won't work for some reason.

Reply:
`eris.Is(err, eris.Cause(ErrInvalidSignature)) || eris.Is(err, eris.Cause(ErrSystemTransactionRequired))`

have to use like this